lib/ret/storage.ex

```diff
@@ -124,6 +129,25 @@ defmodule Ret.Storage do
     end
   end
 
+  # If an owned file does not have a promotion token associated with it, it can be promoted with any given 
+  # promotion token, including nil.
+  defp check_promotion_token(nil, _promotion_token), do: {:ok}
+
+  defp check_promotion_token(actual_promotion_token, promotion_token) do
```

Comment:
you might be able to break this up into a few functions using guard clauses (`when actual_promotion_token == promotion_token` etc). nbd though